riscv/mmu.cc

```diff
@@ -548,3 +535,50 @@ void mmu_t::register_memtracer(memtracer_t* t)
   flush_tlb();
   tracer.hook(t);
 }
+
+reg_t mmu_t::pte_load(reg_t pte_paddr, reg_t addr, bool virt, access_type trap_type, int ptesize) {
+  if (!pmp_ok(pte_paddr, ptesize, LOAD, PRV_S))
+    throw_access_exception(virt, addr, trap_type);
+
+  void* host_pte_paddr = sim->addr_to_mem(pte_paddr);
+  if (ptesize == 4) {
```

Comment:
Would it reduce code duplication here to make `pte_load` type-parameterized on the PTE data type, and going back to the old scheme where `mmu_t::walk` writes something like `vm.ptesize == 4 ? pte_load<uint32_t>(...) : pte_load<uint64_t>(...)`?

Reply:
:+1: The templated approach avoids some repetition

Reply:
cool